pkg/dataingestionframework/data/dif_types.go

```diff
@@ -116,21 +116,40 @@ func (e *DIFEntity) Matching(id string) *DIFEntity {
 }
 
 func (e *DIFEntity) AddMetric(metricType string, kind DIFMetricValKind, value float64, key string) {
+	var metricVal *DIFMetricVal
+	var metricKey *string
+	// Only set non-empty key
+	if key != "" {
+		metricKey = &key
+	}
 	meList, found := e.Metrics[metricType]
 	if !found {
-		meList = append(meList, &DIFMetricVal{})
-		e.Metrics[metricType] = meList
-	}
-	if len(meList) < 1 {
-		return
+		// This is a new metric type
+		metricVal = &DIFMetricVal{Key: metricKey}
+		e.Metrics[metricType] = append(e.Metrics[metricType], metricVal)
```

Comment:
Nit: Is `e.Metrics[metricType]` an empty slice in this case?  So probably doesn't need to use append at all?